### PR TITLE
fix(form-textarea): improve auto-row height calculation timing (closes #3103)

### LINKED ISSUE #3103: <b-form-textarea>: not resizing reactively
When using `rows` and `max-rows` to make a b-form-textarea resize automatically, it does so when typing, but not when updating the model programmatically. 

See this codepen: https://codepen.io/anon/pen/KYXxLY 

When clicking the "Add text" button, the model of the b-form-textarea is updated with a long string, but the textarea isn't resized. If you then type something in the textarea, it resizes correctly.

Some workarounds are available, such as using a key-attribute on the b-form-textarea that is updated on each change, forcing a rerender of the component, or setting a bogus value in the model, and updating it again with this.$nextTick. 

It would be preferable if the component resized itself upon changes in the model, though.

- Operating system and version (Windows, Mac OS X, Android, iOS, Win10 Mobile)
Windows 10
- Browser and version (Chrome, Firefox, Safari, IE, MS Edge, Opera 15+, Android Browser)
Firefox, Chrome
- Version of BootstrapVue that you are using
2.0.0-rc.18
- Reduced test case
https://codepen.io/anon/pen/KYXxLY 

*Edit*:
I updated the codepen to use the latest version of BootstrapVue (see comment below, https://github.com/bootstrap-vue/bootstrap-vue/issues/3103#issuecomment-483243915)

### PR DESCRIPTION
**Pull request by tmorehouse**
## Description of Pull Request:

Switch how the computed height calculation is triggered, by delaying calculation in a nextTick

Fixes #3103

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Enhancement to an existing feature
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e.
      `fixes #xxxx[,#xxxx]`, where "xxxx" is the issue number)
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug
      and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the
      [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e.
      "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos",
      "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated
      from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and
      event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or
      keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a
      suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/form-textarea/form-textarea.js [hunk 1]
```diff
@@ -47,7 +47,8 @@ export default Vue.extend({
   },
   data() {
     return {
-      dontResize: true
+      dontResize: true,
+      heightPx: null
```
Comment (jackmu95) [inline src/components/form-textarea/form-textarea.js]: `heightInPx`

File: src/components/form-textarea/form-textarea.spec.js [hunk 2]
```diff
@@ -848,18 +848,21 @@ describe('form-textarea', () => {
 
     // Check that the internal dontResize flag is now false
     await keepalive.vm.$nextTick()
+    await new Promise(resolve => requestAnimationFrame(resolve))
```
Comment (jackmu95) [inline src/components/form-textarea/form-textarea.spec.js]: Use the new test utils.

File: src/components/form-textarea/form-textarea.spec.js [hunk 3]
```diff
@@ -848,18 +848,21 @@ describe('form-textarea', () => {
...
     expect(textarea.vm.dontResize).toEqual(false)
 
     // v-if the component out of document
     keepalive.setProps({ show: false })
     // Check that the internal dontResize flag is now true
     await keepalive.vm.$nextTick()
+    await new Promise(resolve => requestAnimationFrame(resolve))
```
Comment (jackmu95) [inline src/components/form-textarea/form-textarea.spec.js]: See above.

File: src/components/form-textarea/form-textarea.spec.js [hunk 4]
```diff
@@ -848,18 +848,21 @@ describe('form-textarea', () => {
...
     expect(textarea.vm.dontResize).toEqual(true)
 
     // v-if the component out of document
     keepalive.setProps({ show: true })
     // Check that the internal dontResize flag is now false
     await keepalive.vm.$nextTick()
+    await new Promise(resolve => requestAnimationFrame(resolve))
```
Comment (jackmu95) [inline src/components/form-textarea/form-textarea.spec.js]: See above.

File: src/components/form-textarea/form-textarea.spec.js (hunk 2 above)
Comment (tmorehouse) [reply src/components/form-textarea/form-textarea.spec.js]: Actually not needed any more... I'll remove the await

**Pull request merged**